pennylane/wires.py

```diff
@@ -0,0 +1,369 @@
+# limitations under the License.
+"""
+This module contains the :class:`Wires` class, which takes care of wire bookkeeping.
+"""
+from collections import Sequence, Iterable
+import numpy as np
+from numbers import Number
+
+
+class WireError(Exception):
+    """Exception raised by a :class:`~.pennylane.wires.Wire` object when it is unable to process wires.
+    """
+
+
+def _process(wires):
+    """Converts the input to a tuple of numbers or strings."""
+
+    if isinstance(wires, Wires):
+        # if input is already a Wires object, just copy its wire tuple
+        return wires.wire_tuple
+
+    elif isinstance(wires, Iterable) and all(isinstance(w, Wires) for w in wires):
+        # if the elements are themselves Wires objects, merge them to a new one
+        wires = [w for wires_ in wires for w in wires_.tolist()]
+        return tuple(wires)
+
+    elif isinstance(wires, Iterable) and all(isinstance(w, str) or isinstance(w, Number) for w in wires):
+        # if the elements are strings or numbers, turn iterable into tuple
+        return tuple(wires)
+
+    elif isinstance(wires, Number):
+        # if the input is a single number, interpret as a single wire
+        return (wires,)
+
+    else:
+        raise WireError(
+            "Wires must be represented by a number or string; got {} of type {}.".format(
+                wires, type(wires)
+            )
+        )
+
+
+class Wires(Sequence):
+    """
+    A bookkeeping class for wires, which are ordered collections of unique objects. The i'th object
+    addresses the i'th quantum subsystem.
+
+    There is no conceptual difference between registers of multiple wires and single wires,
+    which are just wire registers of length one.
+
+    Indexing and slicing this sequence will return another ``Wires`` object.
+
+    Args:
+         wires (Any): If iterable, interpreted as an ordered collection of unique objects representing wires.
+            Any other input type is converted into an iterable of a single entry, and hence interpreted as a
+            single wire.
+    """
+
+    def __init__(self, wires):
+
+        self.wire_tuple = _process(wires)
+
+        # check that all wires are unique
+        if len(set(self.wire_tuple)) != len(self.wire_tuple):
+            raise WireError("Wires must be unique; got {}.".format(wires))
+
+    def __getitem__(self, idx):
+        """Method to support indexing. Returns a Wires object representing a register with a single wire."""
+        return Wires(self.wire_tuple[idx])
+
+    def __len__(self):
+        """Method to support ``len()``."""
+        return len(self.wire_tuple)
+
+    def __contains__(self, item):
+        """Method checking if Wires object contains an object."""
+        if isinstance(item, Wires):
+            # If all wires can be found in this object, return True
+            if all(wire in self.wire_tuple for wire in item.wire_tuple):
+                return True
```

Comment:
If the if condition is not met (e.g., not all wires are found in self), then `None` will be returned, shouldn't this return `False`?

This could be fixed by simply making it
```suggestion
            return all(wire in self.wire_tuple for wire in item.wire_tuple)
```
(since this returns `True` iff the if condition evaluates to `True`, `False` otherwise)

Reply:
Did that, thanks!